openmmtools/multistate/multistatesampler.py

```diff
@@ -1774,10 +1775,16 @@ def _update_timing(self, iteration_time, partial_total_time, run_initial_iterati
     def _display_cuda_devices():
         """Query system nvidia-smi to get available GPUs indices and names in debug log."""
         # Read nvidia-smi query, should return empty strip if no GPU is found.
-        cuda_query_output = os.popen("nvidia-smi --query-gpu=index,gpu_name --format=csv,noheader").read().strip()
-        # Split by line jump and comma
-        cuda_devices_list = [entry.split(',') for entry in cuda_query_output.split('\n')]
-        logger.debug(f"CUDA devices available: {*cuda_devices_list,}")
+        cuda_query_output = subprocess.run("nvidia-smi --query-gpu=gpu_uuid,gpu_name,compute_mode  --format=csv", shell=True, capture_output=True, text=True)
+        print(cuda_query_output)
+        if cuda_query_output.stdout:
+            # Split by line jump and comma
+            cuda_devices_list = [entry for entry in cuda_query_output.stdout.splitlines()]
+            logger.debug(f"CUDA devices available: {*cuda_devices_list,}")
+            if "Default" in cuda_query_output.stdout:
```

Comment:
I guess you mean to be checking for `E. Process` or not `Default` here, instead.

Reply:
Oooops, that was because I couldn't get my card to simulate `E. process` since I only have one card in this laptop, but I wanted to make sure the logic, will update that when I test this on a box I put the gpu in `E. process` mode.